<?xml version="1.0"?>
<launch>

 <arg name="joy_dev" default="/dev/input/js0"/>
 <arg name="autorepeat_rate" default="0.0"/>
 <arg name="pad_model" default="ps3"/>
 <arg name="cmd_topic_vel" default="pad_teleop/cmd_vel"/>
 <arg name="cmd_service_set_mode" default="summit_xl_controller/set_mode"/>
 <arg name="cmd_topic_ptz" default="axis_camera/ptz_command"/>
 <arg name="cmd_service_io" default="modbus_io/write_digital_output"/>

 <!-- joy node -->
  <node pkg="joy" type="joy_node" name="joy" output="screen" respawn="true">
    <param name="dev" type="string" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.12" />
    <param name="autorepeat_rate" value="$(arg autorepeat_rate)" type="double" />
  </node>
  
  <!-- Parameters -->
  <param name="cmd_topic_vel" value="$(arg cmd_topic_vel)" type="string"/>
  <param name="cmd_service_io" value="$(arg cmd_service_io)" type="string"/>
  <param name="cmd_topic_ptz" value="$(arg cmd_topic_ptz)" type="string"/>
  <param name="cmd_service_set_mode" value="$(arg cmd_service_set_mode)" type="string"/>
  <rosparam command="load" file="$(find summit_xl_pad)/config/$(arg pad_model).yaml" />
  <node name="summit_xl_pad" pkg="summit_xl_pad" type="summit_xl_pad" output="screen"/>
  
</launch>

